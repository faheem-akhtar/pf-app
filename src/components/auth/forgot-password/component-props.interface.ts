export interface AuthForgotPasswordComponentPropsInterface {
  /**
   * On close auth modal
   */
  onClose: () => void;

  /**
   * On success
   */
  onSuccess: () => void;

  /**
   * On registration link clicked
   */
  onRegister: () => void;
}
