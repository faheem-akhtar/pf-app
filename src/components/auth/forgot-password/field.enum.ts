export enum AuthForgotPasswordFieldEnum {
  /**
   * Email
   */
  email = 'email',
}
