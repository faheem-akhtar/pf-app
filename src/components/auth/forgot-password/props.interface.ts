export interface AuthForgotPasswordPropsInterface {
  /**
   * On close auth modal
   */
  onClose: () => void;

  /**
   * On success
   */
  onSuccess: () => void;

  /**
   * On registration link clicked
   */
  onRegister: () => void;
}
