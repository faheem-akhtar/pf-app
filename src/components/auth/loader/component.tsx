import { domClassMerge } from 'helpers/dom/class-merge';

import styles from './auth-loader.module.scss';

export const AuthLoaderComponent = ({
  isEnabled,
  isCentered,
}: {
  isEnabled: boolean;
  isCentered: boolean;
}): JSX.Element | null =>
  isEnabled ? (
    <div className={styles.loader1}>
      <div
        className={domClassMerge(styles['loader1__spinner'], isCentered ? styles['loader1__spinner--centered'] : '')}
      >
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--1'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--2'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--3'])} />
      </div>
    </div>
  ) : null;
