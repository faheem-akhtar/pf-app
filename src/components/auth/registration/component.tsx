import { useState } from 'react';
import { useTranslationHook } from 'helpers/hook/translation.hook';

import { ApiFetcherResultFailureInterface } from 'api/fetcher-result-failure.interface';
import { AuthRegisterService } from 'services/auth/register.service';
import { AuthRegistrationFieldEnum } from 'components/auth/registration/field.enum';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { CheckboxTemplate } from 'library/checkbox/template';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google-recaptcha/service';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

const captchaService = new GoogleRecaptchaService();

export const AuthRegistrationComponent = ({ close }: { close: () => void }): JSX.Element => {
  const { t } = useTranslationHook();

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');
  const [optedIn, setOptedIn] = useState(false);
  const [errors, setErrors] = useState<Partial<Record<AuthRegistrationFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthRegistrationFieldEnum.email]: [
      validationRequired(t('Enter your email')),
      validationEmail(t('Please enter a valid email')),
    ],
    [AuthRegistrationFieldEnum.password]: [validationRequired(t('Please enter a password'))],
    [AuthRegistrationFieldEnum.firstName]: [validationRequired(t('Please enter a valid first and last name'))],
    [AuthRegistrationFieldEnum.lastName]: [validationRequired(t('Please enter a valid first and last name'))],
  }));

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(setErrors, validators);

  // TODO-FE[CX-219] - Update template
  return (
    <>
      {errorMessage && <div className='authentication__error--general'>{errorMessage}</div>}
      <form
        className='authentication__form'
        onSubmit={(e): void => {
          e.preventDefault();
          if (
            validate({
              [AuthRegistrationFieldEnum.email]: email,
              [AuthRegistrationFieldEnum.password]: password,
              [AuthRegistrationFieldEnum.firstName]: firstName,
              [AuthRegistrationFieldEnum.lastName]: lastName,
            })
          ) {
            captchaService.execute().then((captcha_token) => {
              AuthRegisterService({
                email,
                password,
                first_name: firstName,
                last_name: lastName,
                opted_in: optedIn,
                captcha_token,
              }).then((e) => {
                if (!(e as ApiFetcherResultFailureInterface).ok) {
                  setErrorMessage(
                    (e as ApiFetcherResultFailureInterface).error.body ||
                      t('Something went wrong! Please try again later')
                  );
                  captchaService.reset();
                } else {
                  // Reset server error message
                  setErrorMessage('');
                  // Close modal
                  close();
                }
              });
            });
          }
          return;
        }}
      >
        <div className='authentication__input-area'>
          <div className='authentication__input-label'>{'Enter your email'}</div>
          <input
            type='text'
            className='authentication__input'
            value={email}
            onChange={(e): void => {
              const value = e.target.value;
              validate({
                [AuthRegistrationFieldEnum.email]: value,
              });
              setEmail(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.email]}</div>
        </div>
        <div className='authentication__input-area'>
          <div className='authentication__input-label'>{'Enter your first name'}</div>
          <input
            type='text'
            className='authentication__input'
            value={email}
            onChange={(e): void => {
              const value = e.target.value;
              validate({
                [AuthRegistrationFieldEnum.firstName]: value,
              });
              setFirstName(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.firstName]}</div>
        </div>
        <div className='authentication__input-area'>
          <div className='authentication__input-label'>{'Enter your last name'}</div>
          <input
            type='text'
            className='authentication__input'
            value={email}
            onChange={(e): void => {
              const value = e.target.value;
              validate({
                [AuthRegistrationFieldEnum.lastName]: value,
              });
              setLastName(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.lastName]}</div>
        </div>
        <div className='authentication__input-area'>
          <input
            type='password'
            className='authentication__input'
            value={password}
            onChange={(e): void => {
              const value = e.target.value;
              !validate({
                [AuthRegistrationFieldEnum.password]: value,
              });
              setPassword(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.password]}</div>
        </div>
        <div className='authentication__input-area'>
          <CheckboxTemplate
            id='Opten in'
            checked={optedIn}
            onChange={(e): void => {
              setOptedIn(!!e.target.value);
            }}
          >
            {t('filters-modal/commercial-only')}
          </CheckboxTemplate>
        </div>

        <div className='authentication__form-submit'>
          <ButtonTemplate type='submit' componentType={ButtonComponentTypeEnum.tertiary} size={ButtonSizeEnum.small}>
            {'Register'}
          </ButtonTemplate>
        </div>
      </form>
    </>
  );
};
