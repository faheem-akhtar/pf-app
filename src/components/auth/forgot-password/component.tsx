import React, { useState } from 'react';

import { AuthForgotPasswordFieldEnum } from 'components/auth/forgot-password/field.enum';
import { AuthForgotPasswordPropsInterface } from 'components/auth/forgot-password/props.interface';
import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { AuthResetPasswordService } from 'services/auth/reset-password.service';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ErrorMessageComponent } from 'components/error-message/component';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google/recaptcha.service';
import { InputBaseComponent } from 'library/input/base/component';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import styles from '../auth.module.scss';

export const AuthForgotPasswordComponent = (props: AuthForgotPasswordPropsInterface): JSX.Element => {
  const captchaService = GoogleRecaptchaService();
  const { t } = useTranslation();
  const [email, setEmail] = useState('');
  const [errors, setErrors] = useState<Partial<Record<AuthLoginFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthLoginFieldEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
  }));
  const [isLoading, setIsLoading] = useState(false);

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(errors, setErrors, validators);

  // TODO-FE[CX-499] - Add success message
  return (
    <div data-testid='AuthForgotPasswordComponent'>
      <div className={styles.heading}>{t('auth/forgot-password')}?</div>
      <AuthLoaderComponent isEnabled={isLoading} isCentered={true} />
      {errorMessage && <ErrorMessageComponent message={errorMessage} />}
      <form
        onSubmit={(e): void => {
          e.preventDefault();

          if (
            !validate({
              [AuthForgotPasswordFieldEnum.email]: email,
            })
          ) {
            setIsLoading(true);
            captchaService.execute().then((captcha_token) => {
              AuthResetPasswordService({
                email,
                captcha_token,
              })
                .then((e) => {
                  if (e.ok) {
                    // Reset server error message
                    setErrorMessage('');
                    // Close modal
                    props.onClose();
                    props.onSuccess();
                  } else {
                    setErrorMessage(e.error?.body || `${t('auth/something-wrong')}! ${t('auth/try-later')}`);
                    captchaService.reset();
                  }
                })
                .finally(() => setIsLoading(false));
            });
          }
          return;
        }}
      >
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('email')}
            type='email'
            value={email}
            error={!!errors[AuthForgotPasswordFieldEnum.email]}
            onChange={(value): void => {
              validate({
                [AuthForgotPasswordFieldEnum.email]: value,
              });
              setEmail(value);
            }}
            errorText={errors[AuthForgotPasswordFieldEnum.email]}
          />
        </div>
        <div className={styles['input-area']}>
          <ButtonTemplate
            className={styles.button}
            type='submit'
            componentType={ButtonComponentTypeEnum.primary}
            size={ButtonSizeEnum.regular}
          >
            {t('auth/reset-password')}
          </ButtonTemplate>
        </div>
        <ReCaptchaComponent />
      </form>
      <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
        {t('auth/not-registered-yet')}?&nbsp;{t('auth/create-account')}
      </div>
    </div>
  );
};
