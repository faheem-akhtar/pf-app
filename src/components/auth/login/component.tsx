import React, { useState } from 'react';

import { AuthFacebookService } from 'services/auth/facebook.service';
import { AuthGoogleService } from 'services/auth/google.service';
import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { AuthLoginPropsInterface } from 'components/auth/login/props.interface';
import { AuthLoginService } from 'services/auth/login.service';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ErrorMessageComponent } from 'components/error-message/component';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google-recaptcha/service';
import { IconFacebookTemplate } from 'components/icon/facebook-template';
import { IconGoogleTemplate } from 'components/icon/google-template';
import { InputBaseComponent } from 'library/input/base/component';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import styles from '../auth.module.scss';
const captchaService = new GoogleRecaptchaService();

export const AuthLoginComponent = (props: AuthLoginPropsInterface): JSX.Element => {
  const { t } = useTranslation();

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [errors, setErrors] = useState<Partial<Record<AuthLoginFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthLoginFieldEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
    [AuthLoginFieldEnum.password]: [validationRequired(t('auth/empty-password'))],
  }));
  const [isLoading, setIsLoading] = useState(false);

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(errors, setErrors, validators);

  return (
    <div>
      <div className={styles.heading}>{t('sign-in')}</div>
      {/* Facebook */}
      <ButtonTemplate
        type='button'
        className={domClassMerge(styles.button, styles.facebook)}
        componentType={ButtonComponentTypeEnum.tertiary}
        size={ButtonSizeEnum.small}
        onClick={(): void => {
          setIsLoading(true);
          AuthFacebookService.signIn()
            .then(props.onClose)
            .catch(() => {
              setErrorMessage(`${t('auth/something-wrong')}! ${t('auth/try-later')}`);
              props.onClose();
            })
            .finally(() => setIsLoading(false));
        }}
        icon={{ component: IconFacebookTemplate, position: ButtonIconPositionEnum.left }}
      >
        {t('auth/sign-in-facebook')}
      </ButtonTemplate>
      {/* Google */}
      <ButtonTemplate
        type='button'
        className={domClassMerge(styles.button, styles.google)}
        componentType={ButtonComponentTypeEnum.tertiary}
        size={ButtonSizeEnum.small}
        onClick={(): void => {
          setIsLoading(true);
          AuthGoogleService.signIn()
            .then(props.onClose)
            .catch(() => {
              setErrorMessage(`${t('auth/something-wrong')}! ${t('auth/try-later')}`);
              props.onClose();
            })
            .finally(() => setIsLoading(false));
        }}
        icon={{ component: IconGoogleTemplate, position: ButtonIconPositionEnum.left }}
      >
        {t('auth/sign-in-google')}
      </ButtonTemplate>
      <div className={styles.divider}>
        <span className={styles.text}>{t('auth/or')}</span>
      </div>
      <AuthLoaderComponent isEnabled={isLoading} isCentered={true} />
      {errorMessage && <ErrorMessageComponent message={errorMessage} />}
      <form
        onSubmit={(e): void => {
          e.preventDefault();

          if (
            !validate({
              [AuthLoginFieldEnum.email]: email,
              [AuthLoginFieldEnum.password]: password,
            })
          ) {
            setIsLoading(true);
            captchaService.execute().then((captcha_token) => {
              AuthLoginService({
                email,
                password,
                captcha_token,
              })
                .then((e) => {
                  if (e.ok) {
                    // Reset server error message
                    setErrorMessage('');
                    // Close modal
                    props.onClose();
                  } else {
                    setErrorMessage(e.error.body || `${t('auth/something-wrong')}! ${t('auth/try-later')}`);
                    captchaService.reset();
                  }
                })
                .finally(() => setIsLoading(false));
            });
          }
          return;
        }}
      >
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('email')}
            type='email'
            value={email}
            error={!!errors[AuthLoginFieldEnum.email]}
            onChange={(value): void => {
              validate({
                [AuthLoginFieldEnum.email]: value,
              });
              setEmail(value);
            }}
            errorText={errors[AuthLoginFieldEnum.email]}
          />
        </div>
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('password')}
            type='password'
            value={password}
            error={!!errors[AuthLoginFieldEnum.password]}
            onChange={(value): void => {
              !validate({
                [AuthLoginFieldEnum.password]: value,
              });
              setPassword(value);
            }}
            errorText={errors[AuthLoginFieldEnum.password]}
          />
        </div>
        <div className={styles['input-area']} onClick={props.onForgotPassword}>
          <div className={domClassMerge(styles.link, styles['forgot-password'])}>{t('auth/forgot-password')}?</div>
        </div>
        <div className={styles['input-area']}>
          <ButtonTemplate
            className={styles.button}
            type='submit'
            componentType={ButtonComponentTypeEnum.primary}
            size={ButtonSizeEnum.regular}
          >
            {t('log-in')}
          </ButtonTemplate>
        </div>
        <ReCaptchaComponent />
      </form>
      <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
        {t('auth/not-registered-yet')}?&nbsp;{t('auth/create-account')}
      </div>
    </div>
  );
};
