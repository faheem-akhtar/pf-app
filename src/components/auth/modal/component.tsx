import React, { useState } from 'react';

import { AuthForgotPasswordComponent } from 'components/auth/forgot-password/component';
import { AuthLoginComponent } from 'components/auth/login/component';
import { AuthRegistrationComponent } from 'components/auth/registration/component';
import { AuthScreenEnum } from 'components/auth/screen.enum';
import { IconThickCrossTemplate } from 'components/icon/thick/cross-template';
import { functionNoop } from 'helpers/function/noop';

import styles from './auth-modal-component.module.scss';
import { AuthModalPropsInterface } from './props.interface';

export const AuthModalComponent = ({
  close = functionNoop,
  cancel = functionNoop,
  success = functionNoop,
}: AuthModalPropsInterface): JSX.Element => {
  const [authScreen, setAuthScreen] = useState(AuthScreenEnum.login);
  return (
    <div
      className={styles.content}
      onClick={(e): void => {
        e.stopPropagation();
      }}
    >
      <div className={styles.header}>
        <button
          onClick={(): void => {
            close();
            cancel();
          }}
        >
          <IconThickCrossTemplate class={styles.closeIcon} />
        </button>
      </div>
      {authScreen === AuthScreenEnum.login && (
        <AuthLoginComponent
          onClose={close}
          onSuccess={success}
          onRegister={(): void => setAuthScreen(AuthScreenEnum.registration)}
          onForgotPassword={(): void => setAuthScreen(AuthScreenEnum.forgotPassword)}
        />
      )}
      {authScreen === AuthScreenEnum.registration && (
        <AuthRegistrationComponent
          onClose={close}
          onSuccess={success}
          onLogin={(): void => setAuthScreen(AuthScreenEnum.login)}
        />
      )}
      {authScreen === AuthScreenEnum.forgotPassword && (
        <AuthForgotPasswordComponent
          onClose={close}
          onSuccess={success}
          onRegister={(): void => setAuthScreen(AuthScreenEnum.registration)}
        />
      )}
    </div>
  );
};
