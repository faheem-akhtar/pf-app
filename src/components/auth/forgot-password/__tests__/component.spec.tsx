import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { googleRecaptchaStub } from 'stubs/google/recaptcha.stub';

import { AuthForgotPasswordComponent } from 'components/auth/forgot-password/component';
import * as AuthResetPasswordServiceModule from 'services/auth/reset-password.service';
import * as GoogleRecaptchaServiceModule from 'services/google/recaptcha.service';

import { AuthForgotPasswordPropsInterface } from '../props.interface';

describe('AuthForgotPasswordComponent', () => {
  let props: AuthForgotPasswordPropsInterface;

  beforeEach(() => {
    props = {
      onClose: jest.fn(),
      onRegister: jest.fn(),
      onSuccess: jest.fn(),
    };
  });

  it('renders without throwing any errors', () => {
    const { container } = render(<AuthForgotPasswordComponent {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should submit the form successfully', async () => {
    jest.spyOn(AuthResetPasswordServiceModule, 'AuthResetPasswordService').mockReturnValue(
      Promise.resolve({
        ok: true,
        data: { email: '', captcha_token: '' },
        headers: {} as Headers,
      })
    );

    const googleRecaptchaMock = googleRecaptchaStub();
    jest.spyOn(GoogleRecaptchaServiceModule, 'GoogleRecaptchaService').mockReturnValue(googleRecaptchaMock);

    render(<AuthForgotPasswordComponent {...props} />);
    userEvent.type(screen.getByLabelText('email'), 'email@example.com');
    userEvent.click(screen.getByRole('button', { name: 'auth/reset-password' }));

    expect(await screen.findByTestId('auth-loader'));

    expect(googleRecaptchaMock.execute).toHaveBeenCalledTimes(1);
    expect(googleRecaptchaMock.reset).not.toHaveBeenCalled();

    expect(props.onClose).toHaveBeenCalledTimes(1);
    expect(props.onSuccess).toHaveBeenCalledTimes(1);
  });

  it('should not do anything if email is invalid', () => {
    const googleRecaptchaMock = googleRecaptchaStub();
    jest.spyOn(GoogleRecaptchaServiceModule, 'GoogleRecaptchaService').mockReturnValue(googleRecaptchaMock);

    render(<AuthForgotPasswordComponent {...props} />);
    userEvent.type(screen.getByLabelText('email'), 'invalid email');
    userEvent.click(screen.getByRole('button', { name: 'auth/reset-password' }));

    expect(googleRecaptchaMock.execute).not.toHaveBeenCalled();
    expect(props.onClose).not.toHaveBeenCalled();
    expect(props.onSuccess).not.toHaveBeenCalled();
  });

  it('should call reset while there is error in form submit', async () => {
    jest.spyOn(AuthResetPasswordServiceModule, 'AuthResetPasswordService').mockReturnValue(
      Promise.resolve({
        ok: false,
        error: { body: 'This is an error', url: '', status: 500 },
        headers: {} as Headers,
      })
    );

    const googleRecaptchaMock = googleRecaptchaStub();
    jest.spyOn(GoogleRecaptchaServiceModule, 'GoogleRecaptchaService').mockReturnValue(googleRecaptchaMock);

    render(<AuthForgotPasswordComponent {...props} />);
    userEvent.type(screen.getByLabelText('email'), 'email@example.com');
    userEvent.click(screen.getByRole('button', { name: 'auth/reset-password' }));

    await waitFor(() => {
      expect(googleRecaptchaMock.reset).toHaveBeenCalledTimes(1);
    });

    expect(props.onClose).not.toHaveBeenCalled();
    expect(props.onSuccess).not.toHaveBeenCalled();

    expect(screen.queryByText('auth/something-wrong! auth/try-later')).not.toBeInTheDocument();
    expect(screen.getByText('This is an error')).toBeInTheDocument();
  });

  it('should show custom error when there is no error came from API in form submit', async () => {
    jest.spyOn(AuthResetPasswordServiceModule, 'AuthResetPasswordService').mockReturnValue(
      Promise.resolve({
        ok: false,
        error: { body: '', url: '', status: 500 },
        headers: {} as Headers,
      })
    );

    const googleRecaptchaMock = googleRecaptchaStub();
    jest.spyOn(GoogleRecaptchaServiceModule, 'GoogleRecaptchaService').mockReturnValue(googleRecaptchaMock);

    render(<AuthForgotPasswordComponent {...props} />);
    userEvent.type(screen.getByLabelText('email'), 'email@example.com');
    userEvent.click(screen.getByRole('button', { name: 'auth/reset-password' }));

    await waitFor(() => {
      expect(googleRecaptchaMock.reset).toHaveBeenCalledTimes(1);
    });

    expect(screen.getByText('auth/something-wrong! auth/try-later')).toBeInTheDocument();
  });
});
