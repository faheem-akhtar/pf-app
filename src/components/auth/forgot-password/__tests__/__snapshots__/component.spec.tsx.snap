// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForgotPasswordComponent renders without throwing any errors 1`] = `
<div>
  <div>
    <h1
      class="heading"
    >
      auth/forgot-password
      ?
    </h1>
    
    <form>
      <div
        class="input-area"
      >
        <div
          class="container"
        >
          <div
            class="content"
          >
            <input
              aria-invalid="false"
              aria-label="email"
              class="input"
              id="email"
              name="email"
              type="email"
              value=""
            />
            <div
              class="placeholder"
            >
              <label
                for="email"
              >
                email
              </label>
            </div>
          </div>
        </div>
      </div>
      <div
        class="input-area"
      >
        <button
          class="button primary regular button"
        >
          auth/reset-password
        </button>
      </div>
      <div
        id="recaptcha"
      />
    </form>
    <div
      class="link create-account"
    >
      auth/not-registered-yet
      ?Â 
      auth/create-account
    </div>
  </div>
</div>
`;
